aiohttp_session/__init__.py

```diff
@@ -168,48 +200,57 @@ async def factory(request, handler):
         if session is not None:
             if session._changed:
                 await storage.save_session(request, response, session)
-        if raise_response:
+        if isinstance(response, web.HTTPException):
```

Comment:
`raise_response` is only set to `True` in the try-except block above, the only 'class' of exceptions caught there is `web.HTTPException` see [here](https://github.com/aio-libs/aiohttp-session/blob/1127e80c039a876398ec18fc2c5d448229cf8214/aiohttp_session/__init__.py#L154) so I think this is 1:1 as far as logic is concerned.

Reply:
No, the handler can return `HTTPException` at `response = await handler(request)`.
You'll notice that this is a subclass of `Response` in the 3.x series.

While we encourage people to `raise` an exception instead of returning it, we should ensure backwards compatibility until we move to 4.x.

Reply:
So, if `response = await handler(request)` returns a `HTTPException`, the new code would `raise` it, while the old code would `return` it.

Reply:
Ah, I see. I focused on the wrong part of the change. Thank you :)